tasks/main.yml

```diff
@@ -19,11 +20,11 @@
   when: ansible_pkg_mgr == "apt"
   tags: [postgresql, postgresql-install]
 
-- import_tasks: install_yum.yml
-  when: ansible_pkg_mgr == "yum" and ( ansible_distribution == "RedHat" or ansible_distribution == "CentOS" or ansible_distribution == "OracleLinux")
+- import_tasks: install_rhel.yml
+  when: (ansible_pkg_mgr == "yum" or ansible_pkg_mgr == "dnf") and (ansible_distribution == "RedHat" or ansible_distribution == "CentOS" or ansible_distribution == "OracleLinux" or ansible_distribution == "Rocky")
   tags: [postgresql, postgresql-install]
 
-- import_tasks: install_dnf.yml
+- import_tasks: install_fedora.yml
```

Comment:
Condition here can be just “when distro is fedora”.
Fedora has migrated to dnf many years ago. 
There is no yum on Fedora now.

Reply:
This has been updated